azure-mgmt-network/src/main/java/com/microsoft/azure/management/network/implementation/NetworksImpl.java

```diff
@@ -0,0 +1,135 @@
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for
+ * license information.
+ */
+package com.microsoft.azure.management.network.implementation;
+
+import com.microsoft.azure.CloudException;
+import com.microsoft.azure.Page;
+import com.microsoft.azure.PagedList;
+import com.microsoft.azure.management.network.Network;
+import com.microsoft.azure.management.network.Networks;
+import com.microsoft.azure.management.network.implementation.api.AddressSpace;
+import com.microsoft.azure.management.network.implementation.api.DhcpOptions;
+import com.microsoft.azure.management.network.implementation.api.SubnetInner;
+import com.microsoft.azure.management.network.implementation.api.VirtualNetworkInner;
+import com.microsoft.azure.management.network.implementation.api.VirtualNetworksInner;
+import com.microsoft.azure.management.resources.ResourceGroups;
+import com.microsoft.azure.management.resources.fluentcore.arm.ResourceUtils;
+import com.microsoft.azure.management.resources.fluentcore.utils.PagedListConverter;
+import com.microsoft.azure.management.resources.implementation.api.PageImpl;
+import com.microsoft.rest.RestException;
+import com.microsoft.rest.ServiceResponse;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+class NetworksImpl implements Networks {
+    private final VirtualNetworksInner client;
+    private final ResourceGroups resourceGroups;
+    private final PagedListConverter<VirtualNetworkInner, Network> converter;
+
+    NetworksImpl(final VirtualNetworksInner client, final ResourceGroups resourceGroups) {
+        this.client = client;
+        this.resourceGroups = resourceGroups;
+        this.converter = new PagedListConverter<VirtualNetworkInner, Network>() {
+            @Override
+            public Network typeConvert(VirtualNetworkInner inner) {
+                return createFluentModel(inner);
+            }
+        };
+    }
+
+    @Override
+    public PagedList<Network> list() throws CloudException, IOException {
+        ServiceResponse<PagedList<VirtualNetworkInner>> response = client.listAll();
+        return converter.convert(toPagedList(response.getBody()));
```

Comment:
[PagedListConverter.convert(param)](https://github.com/Azure/azure-sdk-for-java/blob/master/azure-mgmt-resources/src/main/java/com/microsoft/azure/management/resources/fluentcore/utils/PagedListConverter.java#L14) requires param to be a PagedList. 

The type of [client.listAll().getBody()](https://github.com/Azure/azure-sdk-for-java/blob/master/azure-mgmt-network/src/main/java/com/microsoft/azure/management/network/implementation/api/VirtualNetworksInner.java#L577) is already a PagedList, so we don't have to invoke toPagedList(..), simply `converter.convert(response.getBody())` should work.

There are some list methods in SDK that returns List instead of PagedList, an example is [`List` in Availabilitysets](
https://github.com/Azure/azure-sdk-for-java/blob/master/azure-mgmt-compute/src/main/java/com/microsoft/azure/management/compute/implementation/api/AvailabilitySetsInner.java#L334), in such cases we need to invoke `toPagedList`